### PR TITLE
fix: Using GitLab already provided credentials

### PR DESCRIPTION
**Pull request by edeediong**
Signed-off-by: Edidiong Etuk <edidiongmichael08@gmail.com>

### CODE REVIEW

**Review by pst — CHANGES_REQUESTED** [review]
Thanks for your contribution. Using the built in Gitlab environment variables is a good idea. However, here's some review remarks

1. ~If you moved the login to the before script to have the pull during build authenticated, that's a valid goal. But why not simply move it up to be the first item in the script array?~ After writing this, I realized the pull is from Docker Hub. The Gitlab credentials wouldn't help with the limit situation there by being an authenticated pull. So why move the login to `before_script`?
2. You moved away from using `--password-stdin` which I think is the better approach in CI See: https://docs.docker.com/engine/reference/commandline/login/#provide-a-password-using-stdin
3. The docker login registry needs to match the one in the `KBST_DOCKER_IMAGE`, so it would make sense to use the built in variables there too. Optimally even for org and project as well.

**Comment by edeediong** [comment]
> Thanks for your contribution. Using the built in Gitlab environment variables is a good idea. However, here's some review remarks
> 
> 1. ~If you moved the login to the before script to have the pull during build authenticated, that's a valid goal. But why not simply move it up to be the first item in the script array?~ After writing this, I realized the pull is from Docker Hub. The Gitlab credentials wouldn't help with the limit situation there by being an authenticated pull. So why move the login to `before_script`?
> 2. You moved away from using `--password-stdin` which I think is the better approach in CI See: https://docs.docker.com/engine/reference/commandline/login/#provide-a-password-using-stdin
> 3. The docker login registry needs to match the one in the `KBST_DOCKER_IMAGE`, so it would make sense to use the built in variables there too. Optimally even for org and project as well.

Hi @pst thanks for the review. 

1. Here are my thoughts on this:
    - I put the `docker login` in `before_script` because I assumed that it'll make sense to authenticate before building image. What if the base image in the _Dockerfile_ isn't the open-sourced version? However, having looked at [GitLab's documentation](https://docs.gitlab.com/ee/ci/yaml/#before_script) of `before_script`, it doesn't totally make sense to separate them. This is because we concatenate `before_script` and `script` and the former is the script on top. They both run in the same shell. 
    - However, the pull isn't from Docker Hub, it's from GitLab's own Container Registry.
    - I will put the login in the `script`. Let's avoid complications on the project!
2. Thanks for sharing the link that supports my implementation. It'll come in handy during in-house meetings, lol.
3. You're absolutely right! `KBST_DOCKER_IMAGE` can use built-in variables from GitLab and I'll fix that!

File: gitlab/.gitlab-ci.yml [hunk 1]
```diff
@@ -30,8 +30,8 @@ build:
   variables:
     DOCKER_BUILDKIT: 1
   script:
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
```
Comment (pst) [inline gitlab/.gitlab-ci.yml]: Can't merge this. It's still overwriting the `--password-stdin`.

Comment (edeediong) [reply gitlab/.gitlab-ci.yml]: Oh I'm sorry. Let me fix this asap!

Comment (edeediong) [reply gitlab/.gitlab-ci.yml]: Please review @pst

**Review by pst — APPROVED** [review]
LGTM

**Pull request merged**